### PR TITLE
Bug 1721637: return post-formatting hashes for `formatted_replacement…

### PR DESCRIPTION
**Pull request by cgsheeh**
…s` field

At the moment the `formatted_replacements` field is actually the
pre-formatting changeset hashes instead of the post-formatting
hashes, making the value mostly meaningless. Change the value
to the post-formatting hashes to make the value more useful.

### CODE REVIEW

File: landoapi/hg.py [hunk 1]
```diff
@@ -391,14 +391,21 @@ def format(self) -> Optional[List[str]]:
...
 
-        formatted_replacements = check_fix_output_for_replacements(fix_output)
-        logger.info(
-            f"revisions were reformatted: {', '.join(formatted_replacements)}"
-            if formatted_replacements
-            else "autoformatting did not change any revisions"
+        # Exit if no revisions were reformatted.
+        pre_formatting_hashes = check_fix_output_for_replacements(fix_output)
+        if not pre_formatting_hashes:
+            logger.info("autoformatting did not change any revisions")
```
Comment (mitchhentges) [inline landoapi/hg.py:397]: Unsure if this log line is valuable, but that's up to you.

Comment (mitchhentges) [inline landoapi/hg.py:397]: Unsure if this log line is valuable, but that's up to you.

File: landoapi/hg.py [hunk 2]
```diff
@@ -391,14 +391,21 @@ def format(self) -> Optional[List[str]]:
...
+            return None
+
+        post_formatting_hashes = (
+            self.run_hg(["log", "-r", "stack()", "-T", "{node}\n"])
+            .decode("utf-8")
+            .splitlines()[len(pre_formatting_hashes) - 1 :]
         )
 
-        return formatted_replacements
+        logger.info(f"revisions were reformatted: {', '.join(post_formatting_hashes)}")
```
Comment (mitchhentges) [inline landoapi/hg.py:406]: Note that [the bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1721637#c0) says that you'll print both the pre- and post-formatting hashes. I'm indifferent, and I'm guessing that the pre-hash is sufficiently unhelpful that it's not worth printing.

**Review by mitchhentges — APPROVED** [review]
Neat :+1:

File: landoapi/hg.py (hunk 2 above)
Comment (cgsheeh) [reply landoapi/hg.py:406]: Yes, I thought about it again and decided to just post the node of the commit that would actually land.

File: landoapi/hg.py (hunk 1 above)
Comment (cgsheeh) [reply landoapi/hg.py:397]: I agree it's not super useful - dumping it.

**Pull request merged**